src/main.ts

```diff
@@ -161,67 +173,119 @@ export default class TTSPlugin extends Plugin {
+	removeMenu(): void {
+		this.menuVisible = false;
+		clearInterval(this.statusbarIntervalId);
+		this.seekbar?.remove();
+		this.playButton?.remove();
+		this.pauseButton?.remove();
+		this.stopButton?.remove();
+	}
+
+	async createMenu(): Promise<void> {
+		const getSeekbarBackgroundStyle = (value: number): string => {
+			return 'linear-gradient(to right, var(--interactive-accent) 0%, var(--interactive-accent) ' + value + '%, #fff ' + value + '%, white 100%)';
+		}
+		const updateSeekbar = () => {
+			const value = this.serviceManager.progress;
+			this.seekbar.querySelector('input').value = value.toString();
+			this.seekbar.querySelector('input').style.background = getSeekbarBackgroundStyle(value);
+		};
+		const createPlayButton = () => {
+			this.playButton = this.addStatusBarItem();
+			setIcon(this.playButton, 'play-audio-glyph');
+			this.playButton.onClickEvent(() => {
+				this.serviceManager.resume();
+				this.statusbarIntervalId = this.registerInterval(window.setInterval(updateSeekbar, 1000));
+				this.playButton.remove();
+				createPauseButton();
+			});
+		}
+		const createPauseButton = () => {
+			this.pauseButton = this.addStatusBarItem();
+			setIcon(this.pauseButton, 'pause');
+			this.pauseButton.onClickEvent(() => {
+				clearInterval(this.statusbarIntervalId);
+				this.serviceManager.pause();
+				this.pauseButton.remove();
+				createPlayButton();
+			});
+		}
 
 		const markdownView = this.app.workspace.activeEditor;
 		if (markdownView) {
-			if (window.speechSynthesis.speaking) {
-				menu.addItem((item) => {
-					item
-						.setIcon("play-audio-glyph")
-						.setTitle("Add to playback queue")
-						.onClick((async () => {
-							await this.play(markdownView);
-						}));
+			this.menuVisible = true;
+			if (this.serviceManager.isSpeaking()) {
+				this.playButton = this.addStatusBarItem()
+				setIcon(this.playButton, 'play-audio-glyph');
+				this.playButton.onClickEvent(async () => {
+					await this.play(markdownView);
 				});
 			} else {
 				await this.play(markdownView);
 				return;
 			}
 		}
 
-		if (window.speechSynthesis.speaking) {
-			menu.addItem((item) => {
-				item
-					.setIcon("stop-audio-glyph")
-					.setTitle("Stop")
-					.onClick(async () => {
-						this.serviceManager.stop();
-					});
-			});
+		if (this.serviceManager.isSpeaking()) {
+			this.menuVisible = true;
+			// Seekbar
+			const curProgress = this.serviceManager.progress ? this.serviceManager.progress : 0;
+			this.seekbar = this.addStatusBarItem();
+			const slider = this.seekbar.createEl('input', {type: 'range', attr: {min: '0', max: '100', value: curProgress.toString(), step: '1'}});
+			slider.style.background = getSeekbarBackgroundStyle(curProgress);
+			slider.style.border = "solid 1px var(--interactive-accent)";
+			slider.style.borderRadius = "8px";
+			slider.style.height = "7px";
+			slider.style.width = "200px";
+			slider.style.outline = "none";
+			slider.style.transition = "background 450ms ease-in";
```

Comment:
This should be put into the CSS file, not assigned with JS

Reply:
Moved all the static styles to styles.css